partiql-tests-data/fail/static-analysis/primitives/timestamp-constructor.ion

```diff
@@ -0,0 +1,655 @@
+  statement: "TIMESTAMP '2021-03-01T00:+01:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+// Setting to 62 to take account of leap second, if any implement does care.
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid SECOND field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01T00:00:62'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid SECOND field - negative value",
+  statement: "TIMESTAMP '2021-03-01T00:00:-01'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid SECOND field - plus sign",
+  statement: "TIMESTAMP '2021-03-01T00:00:+01'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH OUT TIMEZONE - RFC 3339 Fomat - invalid Precision - negative value",
+  statement: "TIMESTAMP(-1) '2021-03-01T00:00:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+// TIMESTAMP - WITH TIME ZONE - SQL FORMAT
+{
+  name: "TIMESTAMP - WITH TIMEZONE - SQL Fomat - invalid timezone_hour field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01 00:00:00+24:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH TIMEZONE - SQL Fomat - invalid timezone_hour field - below lower bound",
+  statement: "TIMESTAMP '2021-03-01 00:00:00-24:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH TIMEZONE - SQL Fomat - invalid timezone_minute field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01 00:00:00+00:60'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH TIMEZONE - SQL Fomat - invalid timezone_hour field - below lower bound",
+  statement: "TIMESTAMP '2021-03-01 00:00:00-00:60'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+// TIMESTAMP WITH TIME ZONE - RFC 3339 Format
+{
+  name: "TIMESTAMP - WITH TIMEZONE - RFC 3339 Fomat - invalid timezone_hour field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01T00:00:00+24:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH TIMEZONE - RFC 3339 Fomat - invalid timezone_hour field - below lower bound",
+  statement: "TIMESTAMP '2021-03-01T00:00:00-24:00'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH TIMEZONE - RFC 3339 Fomat - invalid timezone_minute field - above upper bound",
+  statement: "TIMESTAMP '2021-03-01T00:00:00+00:60'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH TIMEZONE - RFC 3339 Fomat - invalid timezone_hour field - below lower bound",
+  statement: "TIMESTAMP '2021-03-01T00:00:00-00:60'",
+  assert: {
+    result: StaticAnalysisFail
+  },
+}
+
+{
+  name: "TIMESTAMP - WITH TIMEZONE - RFC 3339 Fomat - invalid timezone_hour symbol",
```

Comment:
nit: could add a similar test using the SQL format

Reply:
See https://github.com/partiql/partiql-tests/pull/102#discussion_r1265818286